doc/nrf/ug_memfault.rst

```diff
@@ -0,0 +1,207 @@
+
+.. _ug_memfault_error:
+
+Error Tracking
+==============
+
+Memfault captures all faults that happen at any time.
+The crash data is sent to the Memfault cloud for further inspection and analysis.
+The faults are tracked using the following methods.
+
+Reboot reason tracking
+----------------------
+
+The device might reboot due to a crash, a brownout, or a firmware update.
+The reboot events for each device are captured and displayed in the Memfault UI.
+
+See `Memfault: Reboot tracking`_ for detailed documentation.
+
+Error tracking with trace events
+--------------------------------
+
+You can use the trace event method if you want to track errors without capturing full coredump traces (that use less storage) and keep the device up and running.
+The following samples implement a user-defined trace reason for demonstration purposes.
+
+* :ref:`memfault_sample`
+* :ref:`peripheral_mds`
+
+Read the **Error Tracking with trace events** section of these sample documents for more information.
+
+See `Memfault: Error Tracking with Trace Events`_ for detailed documentation.
+Faults are captured and displayed on the **Issues** page of Memfault UI.
+
+RAM-backed stack dump collection (Coredumps)
+--------------------------------------------
+
+The coredumps can be triggered to track the fault.
+The **Coredumps** section of the following sample documents provides more information:
+
+* :ref:`memfault_sample`
+* :ref:`peripheral_mds`
+
+See `Memfault: Coredumps`_ for detailed documentation.
+Faults are captured and displayed on the **Issues** page of the Memfault UI.
+
+.. _ug_memfault_metrics:
+
+Collecting Metrics
+==================
+
+The Memfault collects data from each device and visualizes it in a graph.
+The Memfault SDK integration layer in |NCS| adds the system property metrics.
+There are also some metrics specific to |NCS|.
+See **Metrics** section of the following sample documents for more information:
+
+* :ref:`memfault_sample`
+* :ref:`peripheral_mds`
+
+See `Memfault: Collecting Device Metrics`_ for detailed documentation.
+
+Configuration options
+=====================
+
+There are different configuration options defined for the Memfault SDK in |NCS|.
+The configuration options are defined both in `Memfault-SDK`_ and in the integration layer, which integrates the SDK into |NCS|.
+The configuration options of the integration layer are in :file:`modules/memfault/Kconfig`.
+See the Configuration section of the following library documents for different Memfault SDK configuration options defined in |NCS|:
+
+* :ref:`mod_memfault`
+* :ref:`peripheral_mds`
+
+Applications and samples
+************************
+
+The following application uses the Memfault integration in |NCS|:
+
+* :ref:`asset_tracker_v2` - The debug module of :ref:`asset_tracker_v2` uses `Memfault SDK`_ to track |NCS| specific metrics such as LTE and stack metrics.
+  See :ref:`asset_tracker_v2_debug_module` documentation for more details.
```

Comment:
It could be usefult to mention that the application shows how to use Memfault on a custom transport too, ie not only using Memfault's own HTTPS

Reply:
Had a offline chat with Jan tore, and we agreed to cover this in a separate PR.